mono/utils/mono-merp.c

```diff
@@ -357,162 +331,112 @@ mono_init_merp (const intptr_t crashed_pid, const char *signal, MonoStackHash *h
 	merp->hashes = *hashes;
 }
 
-static gchar *
-mono_merp_fingerprint_payload (const char *non_param_data, const MERPStruct *merp)
+static gboolean
+mono_merp_write_fingerprint_payload (const char *non_param_data, const MERPStruct *merp)
 {
-	JsonWriter writer;
-	mono_json_writer_init (&writer);
-
-	mono_json_writer_object_begin(&writer);
-
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "payload");
-	mono_json_writer_printf (&writer, "%s,\n", non_param_data);
-
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "parameters");
-	mono_json_writer_object_begin(&writer);
-
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "ApplicationBundleId:");
-	mono_json_writer_printf (&writer, "\"%s\",\n", merp->bundleIDArg);
-
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "ApplicationVersion:");
-	mono_json_writer_printf (&writer, "\"%s\",\n", merp->versionArg);
-
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "ApplicationBitness:");
-	mono_json_writer_printf (&writer, "\"%s\",\n", get_merp_bitness (merp->archArg));
-
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "ApplicationName:");
-	mono_json_writer_printf (&writer, "\"%s\",\n", merp->serviceNameArg);
-
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "BlameModuleName:");
-	mono_json_writer_printf (&writer, "\"%s\",\n", merp->moduleName);
-
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "BlameModuleVersion:");
-	mono_json_writer_printf (&writer, "\"%s\",\n", merp->moduleVersion);
-
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "BlameModuleOffset:");
-	mono_json_writer_printf (&writer, "\"0x%x\",\n", merp->moduleOffset);
-
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "ExceptionType:");
-	mono_json_writer_printf (&writer, "\"%s\",\n", get_merp_exctype (merp->exceptionArg));
-
-	mono_json_writer_indent (&writer);
-	mono_json_writer_object_key(&writer, "StackChecksum:");
-	mono_json_writer_printf (&writer, "\"0x%x\",\n", merp->hashes.offset_free_hash);
+	int handle = g_open (merp->crashLogPath, O_TRUNC | O_WRONLY | O_CREAT, merp_file_permissions);
+	if (handle == -1) {
```

Comment:
`g_assertf (handle != -1, "Could not open crash log file at %s", merp->merpFilePath);`